org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/debug/AbstractDebugDelegate.java

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Red Hat Inc. and others.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Andrew Obuchowicz (Red Hat Inc.) 
+ *******************************************************************************/
+package org.eclipse.wildwebdeveloper.debug;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.debug.core.DebugPlugin;
+import org.eclipse.debug.core.ILaunch;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.ILaunchManager;
+import org.eclipse.jface.dialogs.ErrorDialog;
+import org.eclipse.lsp4e.debug.DSPPlugin;
+import org.eclipse.lsp4e.debug.launcher.DSPLaunchDelegate;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.wildwebdeveloper.Activator;
+import org.eclipse.wildwebdeveloper.InitializeLaunchConfigurations;
+
+public class AbstractDebugDelegate extends DSPLaunchDelegate{
+	public static final String PROGRAM = "program"; //$NON-NLS-1$
+	public static final String ARGUMENTS = "runtimeArgs"; //$NON-NLS-1$
+	public static final String CWD = DebugPlugin.ATTR_WORKING_DIRECTORY; //$NON-NLS-1$
+	public static final String ENV = ILaunchManager.ATTR_ENVIRONMENT_VARIABLES;
+	public static final String SOURCE_MAPS = "sourceMaps";
+	public static final String PORT = "port"; //$NON-NLS-1$
+	public static final String REQUEST = "request"; //$NON-NLS-1$
+
+	public void launchWithParameters(ILaunchConfiguration configuration, String mode, ILaunch launch,
+			IProgressMonitor monitor, Map<String, Object> param, File debugAdapter) throws CoreException {
+		try {
+			List<String> debugCmdArgs = Collections.singletonList(debugAdapter.getAbsolutePath());
+
+			DSPLaunchDelegateLaunchBuilder builder = new DSPLaunchDelegateLaunchBuilder(configuration, mode, launch,
+					monitor);
+			builder.setLaunchDebugAdapter(InitializeLaunchConfigurations.getNodeJsLocation(), debugCmdArgs);
```

Comment:
Just realized this line is calling `InitializeLaunchConfigurations.getNodeJsLocation()` - it should probably call a function to get the delegates debug adapter location instead

Reply:
Nevermind.. you need node to run the debug adapter :man_facepalming: